common/src/chain/transaction/input.rs

```diff
@@ -31,6 +31,7 @@ use super::{AccountCommand, AccountOutPoint, AccountSpending, OutPointSourceId,
     serde::Serialize,
     serde::Deserialize,
 )]
+#[serde(tag = "type", content = "content")]
```

Comment:
do we use it somewhere? there is dedicated `RpcTxInput` type

Reply:
Yea, it is used. It is used in the wallet and also in the `get_block_json` RPC call. To get rid of this, we'd need to create the RPC equivalent of `Block` and all its constituent parts, effectively duplicating a fair chunk of `common::chain`. I'd consider that to be out of scope for this PR.

Reply:
there is `RpcBlock` already as well

Reply:
So should we revert this? I thought we only care about content for public api and the wallet can just serialize/deserialize as is

Reply:
Not sure. Is there any harm in keeping it? I think, as discussed, whatever wallet needs internally should eventually be transmitted as hex strings so hopefully it will go one day.

Reply:
Do you have a reasonably simple way of finding out where it should be kept and where to remove it? One heuristic is the presence of `HasValueHint` but it's not completely reliable.